ERCS/erc-7817.md

```diff
@@ -0,0 +1,112 @@
+---
+eip: 7817
+title: dApp Security Policy Standard
+description: Defines a document standard for defining the expected on-chain behavior of a dApp frontend.
+author: Bernard Wagner (bernard-wagner)
+status: Draft
+discussions-to: https://ethereum-magicians.org/t/dapp-security-policy/21431
+type: Meta
+created: 2024-10-22
+---
+
+## Abstract
+
+Introduce a standardized JSON document structure that wallets can interpret to determine whether transaction signing requests should be permitted or rejected. Policy documents should be publicly discoverable using a notary service so that a policy's authenticity can be assumed even when the origin of the transaction signing request may be compromised.
+
+As an initial implementation, DNS-in-ENS can be used as the discovery mechanism as described in ENSIP-TBC.
+
+## Motivation
+
+Hackers often target dApp front-ends to coerce users into signing transactions that allow the hacker to transfer victims' funds. By introducing a security policy standard, wallet providers can implement safeguards to protect users against such attacks.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Policy Document
+
+The document MUST be in JSON format and follow the structure defined below:
+
+```json
```

Comment:
I'd recommend using a JSON Schema to define your document structure. See https://json-schema.org/